### PR TITLE
Trigger NetStreamAppendBytesAction.END_SEQUENCE when the mediasource buf...

### PR DESCRIPTION
**Pull request by bclwhitaker**
...fers empty. Requires a corresponding pull in videojs-contrib-media-sources.

### CODE REVIEW

File: src/videojs-hls.js [hunk 1]
```diff
@@ -267,9 +267,16 @@ videojs.Hls.prototype.handleSourceOpen = function() {
...
     player = this.player(),
-    sourceBuffer = this.mediaSource.addSourceBuffer('video/flv; codecs="vp6,aac"');
+    mediaSource = this.mediaSource,
+    sourceBuffer = mediaSource.addSourceBuffer('video/flv; codecs="vp6,aac"');
 
   this.sourceBuffer = sourceBuffer;
+
+  // signal whenever the buffer flushes to the swf
+  this.sourceBuffer.addEventListener('updateend', function() {
+    mediaSource.endOfStream();
```
Comment (dmlap) [inline src/videojs-hls.js]: Won't this trigger endOfStream() after every segment finishes?

File: src/videojs-hls.js [hunk 2]
```diff
@@ -671,7 +678,6 @@ videojs.Hls.prototype.loadSegment = function(segmentUri, offset) {
       offset: offset,
       bytes: new Uint8Array(this.response)
     });
-    player.trigger('progress');
```
Comment (dmlap) [inline src/videojs-hls.js]: Why did you remove this?

File: test/videojs-hls_test.js [hunk 3]
```diff
@@ -1734,7 +1733,7 @@ test('calls ended() on the media source at the end of a playlist', function() {
   requests[0].response = new ArrayBuffer(17);
   requests.shift().respond(200, null, '');
 
-  strictEqual(endOfStreams, 1, 'ended media source');
+  strictEqual(updateendListener, 1, 'ended media source');
```
Comment (dmlap) [inline test/videojs-hls_test.js]: `updateend` should fire after every appendBuffer() operation has completed, not the end of the stream. I think this test is asserting on the wrong thing now.

File: src/videojs-hls.js (hunk 1 above)
Comment (bclwhitaker) [inline src/videojs-hls.js]: Yes, it was.  Most of this logic has now been moved into the mediaSources project via https://github.com/videojs/videojs-contrib-media-sources/pull/24

File: src/videojs-hls.js (hunk 2 above)
Comment (bclwhitaker) [inline src/videojs-hls.js]: progress events are already firing from videojs because the bufferedPercent is changing.  Since all the other changes in this pull has been moved into mediaSources I opened a separate pull request to remove just this event #252.  Firing progress here was sometimes causing the buffered region in the player to misreport.
